pkgs/top-level/default.nix

```diff
@@ -23,7 +23,7 @@
   localSystem
 
 , # The system packages will ultimately be run on.
-  crossSystem ? localSystem
+  crossSystem
```

Comment:
I can imagine someone, who doesn't use `<nixpkgs/default.nix>`, but uses `<nixpkgs/pkgs/top-level/default.nix>` to remove all the impurities, introduced in `impure.nix`. Previously everything except `localSystem` had defaults.

Reply:
The funny thing is that if you didn't explicitly supply it before it woudn't actually work because of the congruence "feature". It would break `stdenv` as described in the comment.

Reply:
I'm fine with exporting `localSystem` and `crossSystem` to `config`, but overriding those in `config`... Not sure it is worth it. If smbd will ever want to `config.localSystem != config.crossSystem` we can instead export `isCrossBuild` for this usecase.